pkg/cache/tas_flavor_snapshot.go

```diff
@@ -223,6 +224,7 @@ func (s *TASFlavorSnapshot) addTASUsage(domainID utiltas.TopologyDomainID, usage
 		s.leaves[domainID].tasUsage = resources.Requests{}
 	}
 	s.leaves[domainID].tasUsage.Add(usage)
+	s.leaves[domainID].tasUsage.Add(resources.Requests{corev1.ResourcePods: 1})
```

Comment:
I think this is not accurate. For task workloads the number of pods in a domain needs to be taken from the assignment. I think to propagate this information we need to adjust a couple of structures, probably starting from `TopologyDomainRequests`.

Another alternative is to just iterate over all pods, not just non-TAS, and then update the snapshot. It is much easier in practice, but would require as too cache all pods.

Reply:
Oh, that sounds correct. I will refine the`TopologyDomainRequests` to store the pod count.
Thanks.

Reply:
@mimowo I addressed this here: https://github.com/kubernetes-sigs/kueue/pull/4271/commits/4653eccd3468cb8727b430850595c038af191d63

Thank you again for great catch :)